Build failure on Windows

```
Preprocessing library for system-fileio-0.3.16.5..
Building library for system-fileio-0.3.16.5..
[1 of 1] Compiling Filesystem       ( lib\Filesystem.hs, dist\build\Filesystem.o )

lib\Filesystem.hs:720:5: error:
Error:     parse error (possibly incorrect indentation or mismatched brackets)
 |
 |     (toInteger (Win32.wHour stime) * 3600) +
 |     ^
```
This happens regardless of ghc version (tested: 8.4 up to 9.10).
Here is a patch that fixes the problem:
```
--- a/system-fileio/lib/Filesystem.hs
+++ b/system-fileio/lib/Filesystem.hs
@@ -711,18 +711,21 @@ getModified path = do
   let ftime = Win32.bhfiLastWriteTime info
   stime <- Win32.fileTimeToSystemTime ftime
 
-  let date = fromGregorian
-    (fromIntegral (Win32.wYear stime))
-    (fromIntegral (Win32.wMonth stime))
-    (fromIntegral (Win32.wDay stime))
-
-  let seconds = secondsToDiffTime $
-    (toInteger (Win32.wHour stime) * 3600) +
-    (toInteger (Win32.wMinute stime) * 60) +
-    (toInteger (Win32.wSecond stime))
-
-  let msecs = picosecondsToDiffTime $
-    (toInteger (Win32.wMilliseconds stime) * 1000000000)
+  let date =
+        fromGregorian
+          (fromIntegral (Win32.wYear stime))
+          (fromIntegral (Win32.wMonth stime))
+          (fromIntegral (Win32.wDay stime))
+
+  let seconds =
+        secondsToDiffTime $
+          (toInteger (Win32.wHour stime) * 3600) +
+          (toInteger (Win32.wMinute stime) * 60) +
+          (toInteger (Win32.wSecond stime))
+
+  let msecs =
+        picosecondsToDiffTime $
+          (toInteger (Win32.wMilliseconds stime) * 1000000000)
 
   return (UTCTime date (seconds + msecs))
 #else
```

--- a/.github/workflows/stack.yml
+++ b/.github/workflows/stack.yml
@@ -1,0 +1,66 @@
+name: stack
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  stack:
+    name: Stack ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    timeout-minutes: 60
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+
+    env:
+      stack: 'stack --system-ghc --no-install-ghc'
+
+    steps:
+
+    - uses: actions/checkout@v4
+
+    - name: Install GHC and stack with haskell-actions/setup
+      uses: haskell-actions/setup@v2
+      id:   setup
+      with:
+        ## Keep the GHC version in sync with stack.yaml
+        ghc-version: 9.8.2
+        enable-stack: true
+        cabal-update: false
+
+    - name: Restore cached dependencies
+      uses: actions/cache/restore@v4
+      id:   cache
+      with:
+        path: |
+          ${{ steps.setup.outputs.stack-root }}
+        key: ${{ runner.os }}-stack-${{ steps.setup.outputs.stack-version }}-ghc-${{ steps.setup.outputs.ghc-version }}-hash-${{ github.sha }}
+        restore-keys: |
+             ${{ runner.os }}-stack-${{ steps.setup.outputs.stack-version }}-ghc-${{ steps.setup.outputs.ghc-version }}-
+
+    - name: Build dependencies
+      if:   steps.cache.outputs.cache-hit != 'true'
+      run:  ${{ env.stack }} test --dependencies-only
+
+    - name: Build w/ tests
+      run: ${{ env.stack }} test --no-run-tests
+
+    - name: Run tests
+      run: ${{ env.stack }} test
+
+    - name: Cache dependencies
+      if:   always() && steps.cache.outputs.cache-hit != 'true'
+      uses: actions/cache/save@v4
+      with:
+        path: |
+          ${{ steps.setup.outputs.stack-root }}
+        key:  ${{ steps.cache.outputs.cache-primary-key }}

--- a/stack.yaml
+++ b/stack.yaml
@@ -1,4 +1,5 @@
 resolver: nightly-2024-10-21
+# If you update the snapshot, also update .github/workflows/stack.yml to the matching GHC version
 
 packages:
 - system-fileio

--- a/system-fileio/ChangeLog.md
+++ b/system-fileio/ChangeLog.md
@@ -1,5 +1,12 @@
 # Changelog for system-fileio
 
+## 0.3.16.6
+
+* Fix building on Windows ([issue #31](https://github.com/fpco/haskell-filesystem/issues/31), regression in 0.3.16.5).
+* Tested with:
+  - Cabal and GHC 8.0 - 9.12.0 on Linux
+  - Stack and GHC 9.8.2 on Linux, macOS and Windows.
+
 ## 0.3.16.5
 
 * Fix building of the testsuite with `unix-2.8`.

--- a/system-fileio/lib/Filesystem.hs
+++ b/system-fileio/lib/Filesystem.hs
@@ -710,20 +710,17 @@ getModified path = do
   info <- withHANDLE path Win32.getFileInformationByHandle
   let ftime = Win32.bhfiLastWriteTime info
   stime <- Win32.fileTimeToSystemTime ftime
-
-  let date = fromGregorian
-    (fromIntegral (Win32.wYear stime))
-    (fromIntegral (Win32.wMonth stime))
-    (fromIntegral (Win32.wDay stime))
-
-  let seconds = secondsToDiffTime $
-    (toInteger (Win32.wHour stime) * 3600) +
-    (toInteger (Win32.wMinute stime) * 60) +
-    (toInteger (Win32.wSecond stime))
-
-  let msecs = picosecondsToDiffTime $
-    (toInteger (Win32.wMilliseconds stime) * 1000000000)
-
+  let
+    date = fromGregorian
+      (fromIntegral (Win32.wYear stime))
+      (fromIntegral (Win32.wMonth stime))
+      (fromIntegral (Win32.wDay stime))
+    seconds = secondsToDiffTime $
+      (toInteger (Win32.wHour stime) * 3600) +
+      (toInteger (Win32.wMinute stime) * 60) +
+      (toInteger (Win32.wSecond stime))
+    msecs = picosecondsToDiffTime $
+      (toInteger (Win32.wMilliseconds stime) * 1000000000)
   return (UTCTime date (seconds + msecs))
 #else
   stat <- posixStat "getModified" path

